// Copyright 2009 Google Inc. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Protocol buffer used to represent per-resource gzip rule results.

syntax = "proto2";

import "pagespeed/proto/pagespeed_output.proto";

package pagespeed;

message GzipDetails {
  extend ResultDetails {
    // In order to attach it to Result.details
    optional GzipDetails message_set_extension = 18298;  // int('Gz')
  }

  message PerUrlSavings {
    required string url = 1;
    required int32 saved_bytes = 2;
  }

  repeated PerUrlSavings url_savings = 1;
}
